### PR TITLE
Feature/formatting json yaml

### LINKED ISSUE #30063: editorconfig: use an indentation of 2 for json and yaml files
### Summary (*)
Consider using a indentation of 2 for json and yaml files.
The recommended default for the package.json and yaml files is 2.

The only exception is composer that uses 4.

### Examples (*)
_None, this just a formatting suggestion_

### Proposed solution
See my Pr #30062 

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users with no workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [x] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### LINKED ISSUE #30066: [Issue] Feature/formatting json yaml
This issue is automatically created based on existing pull request: magento/magento2#30062: Feature/formatting json yaml

---------

### Description (*)
Consider using a indentation of 2 for json and yaml files.
The recommended default for the package.json and yaml files is 2.
The only exception is composer that uses 4.

I know yaml is not a file format found in Magento 2 by default.
But it is used for some use case.

_This PR suggests adding more rules to the editorconfig._

### Manual testing scenarios (*)
_None, this just a formatting suggestion_

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

**No tests needed**

### PR DESCRIPTION
**Pull request by GrimLink**
### Description (*)
Consider using a indentation of 2 for json and yaml files.
The recommended default for the package.json and yaml files is 2.
The only exception is composer that uses 4.

I know yaml is not a file format found in Magento 2 by default.
But it is used for some use case.

_This PR suggests adding more rules to the editorconfig._

### Fixed Issues (if relevant)

1. Fixes #30063
2. Fixes magento/magento2#30066

### Manual testing scenarios (*)
_None, this just a formatting suggestion_

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

**No tests needed**

### CODE REVIEW

File: auth.json.sample [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
-    "http-basic": {
-        "repo.magento.com": {
-            "username": "<public-key>",
-            "password": "<private-key>"
-        }
+  "http-basic": {
```
Comment (ihor-sviziev) [inline auth.json.sample]: Seems like auth.json should also be 4 spaces as it generated using composer. Isn’t it?

Comment (GrimLink) [reply auth.json.sample]: Seems you are right.
I have update my commit.

**Comment by sidolov** [comment]
@magento create issue

**Comment by GrimLink** [comment]
@sidolov I already made an issue for this PR
See #30063

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Review by ihor-sviziev — APPROVED** [review]
✔ Approved.

Failing tests looks not related to changes form this PR.

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8219](https://jira.corp.magento.com/browse/ENGCOM-8219) has been created to process this Pull Request

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by engcom-Bravo** [comment]
Dev experience is required for testing of this PR. Please note that Manual testing has not been performed.

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

File: .editorconfig [hunk 2]
```diff
@@ -14,8 +14,5 @@ trim_trailing_whitespace = false
 [*.{yml,yaml,json}]
 indent_size = 2
 
-[composer.{json,lock}]
-indent_size = 4
-
-[auth.json]
+[{composer, auth}.json]
```
Comment (ihor-sviziev) [inline .editorconfig:17]: @engcom-Foxtrot basically it's not the same. You just removed option for `composer.lock` files.

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
@engcom-Foxtrot see my comment here https://github.com/magento/magento2/pull/30062#pullrequestreview-501951680

File: .editorconfig (hunk 2 above)
Comment (engcom-Foxtrot) [reply .editorconfig:17]: @ihor-sviziev yes, composer.lock inherits settings from general configuration *. It's not overridden by *.{yml,yaml,json}

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8219](https://jira.corp.magento.com/browse/ENGCOM-8219) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
Dev experience is required for testing this PR. Please note that Manual testing has not been performed.

**Comment by engcom-Foxtrot** [comment]
QA passed.

**Pull request merged**